server/test/socketManager-test.js

```diff
@@ -0,0 +1,338 @@
+const { assert } = require('chai');
+const { Server } = require('bweb');
+const blgr = require('blgr');
+const bsock = require('bsock');
+const { NodeClient, WalletClient } = require('bclient');
+const MultisigClient = require('bmultisig/lib/client');
+const SocketManager = require('../socketManager');
+const { initFullNode, initWallet } = require('./utils/regtest');
+
+const ports = {
+  full: {
+    p2p: 49331,
+    node: 49332,
+    wallet: 49333
+  },
+  spv: {
+    p2p: 49431,
+    node: 49432,
+    wallet: 49433
+  },
+  manager: 1234
+};
+
+const serverApiKey = 'foo';
+
+describe('socketManager', function() {
+  let socketManager,
+    logger,
+    options,
+    apiKey,
+    client,
+    node,
+    nclient,
+    wclient,
+    mineBlocks;
+
+  before(
+    'setup bcoin node, chain, client, and SocketManager',
+    async function() {
+      this.timeout(3000);
+      try {
+        logger = new blgr({
+          level: 'info'
+        });
+        await logger.open();
+        node = await initFullNode({
+          ports,
+          memory: true,
+          logLevel: 'none',
+          apiKey: serverApiKey
+        });
+        nclient = new NodeClient({
+          port: ports.full.node,
+          apiKey: serverApiKey
+        });
+        wclient = new WalletClient({
+          port: ports.full.wallet,
+          apiKey: serverApiKey
+        });
+
+        // setting up the chain by mining some blocks
+        await initWallet(wclient);
+        await wclient.execute('selectwallet', ['test']);
+        const coinbase = await wclient.execute('getnewaddress', ['blue']);
+        const blockCount = 10;
+
+        // make a function so other tests can use this
+        mineBlocks = n => nclient.execute('generatetoaddress', [n, coinbase]);
+        const result = await mineBlocks(blockCount);
+        assert(
+          result && result.length === blockCount,
+          `Chain not initiated properly`
+        );
+
+        // setup manager
+        apiKey = 'foobar';
+        options = {
+          apiKey,
+          port: ports.manager
+        };
+        socketManager = new SocketManager(options);
+
+        await socketManager.open();
+
+        // testing client
+        client = bsock.connect(ports.manager);
+        client.on('error', function(e) {
+          // eslint-disable-next-line no-console
```

Comment:
You could prob disable this for the whole file since its a test if you wanted to

Reply:
I like keeping for now because it forces me to clean up console statements that I don't need. And most other console logs should really be asserts in the test.